cluster-autoscaler/estimator/binpacking_estimator.go

```diff
@@ -51,27 +57,74 @@ func NewBinpackingNodeEstimator(predicateChecker *simulator.PredicateChecker) *B
 // still be maintained.
 // It is assumed that all pods from the given list can fit to nodeTemplate.
 // Returns the number of nodes needed to accommodate all pods from the list.
-func (estimator *BinpackingNodeEstimator) Estimate(pods []*apiv1.Pod, nodeTemplate *schedulernodeinfo.NodeInfo,
+func (estimator *BinpackingNodeEstimator) Estimate(
+	pods []*apiv1.Pod,
+	nodeTemplate *schedulernodeinfo.NodeInfo,
 	upcomingNodes []*schedulernodeinfo.NodeInfo) int {
 	podInfos := calculatePodScore(pods, nodeTemplate)
 	sort.Slice(podInfos, func(i, j int) bool { return podInfos[i].score > podInfos[j].score })
 
-	newNodes := make([]*schedulernodeinfo.NodeInfo, 0)
+	newNodeNames := make([]string, 0)
+
+	if err := estimator.clusterSnapshot.Fork(); err != nil {
+		klog.Errorf("Error while calling ClusterSnapshot.Fork; %v", err)
+		return 0
+	}
+	defer func() {
+		if err := estimator.clusterSnapshot.Revert(); err != nil {
+			klog.Errorf("Error while calling ClusterSnapshot.Revert; %v", err)
```

Comment:
At this point CA is completely detached from reality. I would consider making it a Fatalf.

Reply:
I am changing reaction to error on `Revert` to Fatalf` for now. 
I am also adding entry to followups to drop returning of errors from `ClusterSnapshot` methods for which it does not make sense.